examples/ecs/parallel_query.rs

```diff
@@ -0,0 +1,64 @@
+use bevy::{prelude::*, tasks::prelude::*};
+use rand::random;
+
+struct Velocity(Vec2);
+
+fn spawn_system(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    commands.spawn(Camera2dComponents::default());
+    let texture_handle = asset_server.load("assets/branding/icon.png").unwrap();
+    for _ in 0..128 {
+        commands
+            .spawn(SpriteComponents {
+                material: materials.add(texture_handle.into()),
```

Comment:
i want to encourage users to reuse materials when they can (as different materials require gpu state changes during rendering). can we hoist this above the for loop like this?

```rust
let material = materials.add(texture_handle.into());
```